src/worker/smartOrderRouter.ts

```diff
@@ -0,0 +1,235 @@
+import {
+  AlphaRouter,
+  AlphaRouterConfig,
+  CachingGasStationProvider,
+  CachingTokenListProvider,
+  EIP1559GasPriceProvider,
+  HeuristicGasModelFactory,
+  ICache,
+  ID_TO_CHAIN_ID,
+  IMetric,
+  MetricLoggerUnit,
+  PoolProvider,
+  QuoteProvider,
+  routeAmountsToString,
+  setGlobalMetric, SwapRoute, UniswapMulticallProvider,
+  URISubgraphProvider
+} from '@uniswap/smart-order-router'
+import * as Comlink from 'comlink'
+import { NETWORK_URLS } from 'connectors/constants'
+import { ethers } from 'ethers'
+import JSBI from 'jsbi'
+import { GetQuoteResult } from 'state/routing/types'
+
+type TokenInRoute = Pick<Token, 'address' | 'chainId' | 'symbol' | 'decimals'>
+
+type PoolInRoute = {
+  type: 'v3-pool'
+  address: string
+  tokenIn: TokenInRoute
+  tokenOut: TokenInRoute
+  sqrtRatioX96: string
+  liquidity: string
+  tickCurrent: string
+  fee: string
+  amountIn?: string
+  amountOut?: string
+}
+
+class MetricLogger extends IMetric {
+  putDimensions() {
+    return
+  }
+
+  putMetric(key: string, value: number, unit?: MetricLoggerUnit) {
+    console.info({ key, value, unit }, `[Metric]: ${key}: ${value} | ${unit ? unit : ''}`)
+  }
+}
+setGlobalMetric(new MetricLogger())
+
+const DEFAULT_ROUTING_CONFIG: AlphaRouterConfig = {
+  topN: 2,
+  topNDirectSwaps: 2,
+  topNTokenInOut: 3,
+  topNSecondHop: 0,
+  topNWithEachBaseToken: 3,
+  topNWithBaseToken: 6,
+  topNWithBaseTokenInSet: false,
+  maxSwapsPerPath: 3,
+  minSplits: 1,
+  maxSplits: 7,
+  distributionPercent: 5,
+}
+
+class Cache<T> implements ICache<T> {
+  async get() {
+    return undefined
+  }
+  async set() {
+    return false
+  }
+  async has() {
+    return false
+  }
+}
+
+const chainId = 1
+const provider = new ethers.providers.JsonRpcProvider(NETWORK_URLS[chainId])
+
+const subgraphProvider = new URISubgraphProvider(chainId, 'https://ipfs.io/ipfs/QmfArMYESGVJpPALh4eQXnjF8HProSF1ky3v8RmuYLJZT4')
+
+const multicall2Provider =
+  new UniswapMulticallProvider(chainId, provider, 375_000)
+
+const poolProvider = new PoolProvider(ID_TO_CHAIN_ID(chainId), multicall2Provider)
+
+const tokenListProvider = new CachingTokenListProvider(chainId, DEFAULT_TOKEN_LIST, new Cache())
+
+const router = new AlphaRouter({
+  chainId,
+  provider,
+  subgraphProvider,
+  multicall2Provider,
+  poolProvider,
+  quoteProvider: new QuoteProvider(
+    chainId,
+    provider,
+    multicall2Provider,
+    {
+      retries: 2,
+      minTimeout: 100,
+      maxTimeout: 1000,
+    },
+    {
+      multicallChunk: 210, // 210
+      gasLimitPerCall: 705_000, // 705
+      quoteMinSuccessRate: 0.15,
+    },
+    {
+      gasLimitOverride: 2_000_000,
+      multicallChunk: 70,
+    }
+  ),
+  gasPriceProvider: new CachingGasStationProvider(chainId, new EIP1559GasPriceProvider(provider), new Cache()),
+  gasModelFactory: new HeuristicGasModelFactory(),
+  blockedTokenListProvider: undefined,
+  tokenProvider: tokenListProvider,
```

Comment:
Using just a tokenListProvider for this means that if the user specifies a token not in the token list, the API will return a 404. Should use a CachingTokenProviderWithFallback (it will check the token list first, then search on-chain if it doesnt find the token, and do caching)

Same for pool provider actually, we should use the caching version

Best to try and match as closely to the config here: https://github.com/Uniswap/routing-api/blob/main/lib/handlers/quote/injector.ts#L204-L286